### PR TITLE
Disable Kerrigan if HotS campaign is disabled

### PR DESCRIPTION
**Pull request by Ziktofel**
Assumes Kerrigan to be not present if HotS campaign is disabled

### CODE REVIEW

File: worlds/sc2/Options.py [hunk 1]
```diff
@@ -381,7 +375,7 @@ class KerriganPresence(Choice):
         In missions where the Kerrigan unit is required, story abilities are given in same way as Grant Story Tech is set to true
     Not Present And No Passives:  In addition to the above, Kerrigan's passive abilities affecting other units (such as Twin Drones) will not appear.
 
-    Note :Recommended to set Not Present if playing Legacy of the Void without Heart of the Swarm due to The Infinite Cycle mission mechanics
+    Note: It's always assumed to "Not Present" if Heart of the Swarm campaign is disabled.
```
Comment (MatthewMarinets) [inline worlds/sc2/Options.py]: Should be
> Always set to "Not Present" if [...]

File: worlds/sc2/Client.py [hunk 2]
```diff
@@ -21,12 +21,16 @@
...
+from worlds.sc2.Options import (MissionOrder, KerriganPrimalStatus, kerrigan_unit_available, KerriganPresence,
+                                GameSpeed,
+                                GenericUpgradeItems, GenericUpgradeResearch, ColorChoice, GenericUpgradeMissions,
+                                LocationInclusion, ExtraLocations, MasteryLocations, ChallengeLocations,
+                                VanillaLocations,
+                                DisableForcedCamera, SkipCutscenes, GrantStoryTech, TakeOverAIAllies, RequiredTactics,
+                                SpearOfAdunPresence,
+                                SpearOfAdunPresentInNoBuild, SpearOfAdunAutonomouslyCastAbilityPresence,
+                                SpearOfAdunAutonomouslyCastPresentInNoBuild, EnableHotsMissions
+                                )
```
Comment (MatthewMarinets) [inline worlds/sc2/Client.py]: (minor): This indentation is kind of gross, a lot of wasted whitespace on the left.

File: worlds/sc2/Client.py [hunk 3]
```diff
@@ -21,12 +21,13 @@
...
-    LocationInclusion, ExtraLocations, MasteryLocations, ChallengeLocations, VanillaLocations,
-    DisableForcedCamera, SkipCutscenes, GrantStoryTech, TakeOverAIAllies, RequiredTactics, SpearOfAdunPresence,
-    SpearOfAdunPresentInNoBuild, SpearOfAdunAutonomouslyCastAbilityPresence, SpearOfAdunAutonomouslyCastPresentInNoBuild
-)
+from worlds.sc2.Options import MissionOrder, KerriganPrimalStatus, kerrigan_unit_available, KerriganPresence, \
+    GameSpeed, GenericUpgradeItems, GenericUpgradeResearch, ColorChoice, GenericUpgradeMissions, \
+    LocationInclusion, ExtraLocations, MasteryLocations, ChallengeLocations, VanillaLocations, \
+    DisableForcedCamera, SkipCutscenes, GrantStoryTech, TakeOverAIAllies, RequiredTactics, SpearOfAdunPresence, \
+    SpearOfAdunPresentInNoBuild, SpearOfAdunAutonomouslyCastAbilityPresence, \
+    SpearOfAdunAutonomouslyCastPresentInNoBuild
```
Comment (MatthewMarinets) [inline worlds/sc2/Client.py:29]: Now where'd the parens go? Just do:
```python
from worlds.sc2.Options import (
    MissionOrder, KerriganPrimalStatus, kerrigan_unit_available, KerriganPresence,
    # ...
)
```
And you never need an EOL backslash again. Is this PyCharm doing autoformatting?

Comment (Ziktofel) [reply worlds/sc2/Client.py:29]: yeah, the autoformating when auto-add import does that formatting, when it's backshalsh'd, the problem doesn't exist

File: worlds/sc2/PoolFilter.py [hunk 4]
```diff
@@ -553,7 +554,8 @@ def filter_items(multiworld: MultiWorld, player: int, mission_req_table: Dict[SC
 def get_used_races(mission_req_table: Dict[SC2Campaign, Dict[str, MissionInfo]], multiworld: MultiWorld, player: int) -> Set[SC2Race]:
     grant_story_tech = get_option_value(multiworld, player, "grant_story_tech")
     take_over_ai_allies = get_option_value(multiworld, player, "take_over_ai_allies")
-    kerrigan_presence = get_option_value(multiworld, player, "kerrigan_presence")
+    kerrigan_presence = get_option_value(multiworld, player, "kerrigan_presence") \
+        and SC2Campaign.HOTS in get_enabled_campaigns(multiworld, player)
```
Comment (MatthewMarinets) [inline worlds/sc2/PoolFilter.py:558]: These checks are no longer necessary if the slot_data was updated earlier.

File: worlds/sc2/PoolFilter.py [hunk 5]
```diff
@@ -132,7 +132,8 @@ def move_mission(mission: SC2Mission, current_pool, new_pool):
             and get_option_value(multiworld, player, "shuffle_campaigns") == ShuffleCampaigns.option_false:
         move_mission(SC2Mission.DARK_WHISPERS, MissionPools.EASY, MissionPools.STARTER)
     # HotS
-    kerriganless = get_option_value(multiworld, player, "kerrigan_presence") not in kerrigan_unit_available
+    kerriganless = get_option_value(multiworld, player, "kerrigan_presence") not in kerrigan_unit_available \
+        or SC2Campaign.HOTS not in enabled_campaigns
```
Comment (MatthewMarinets) [inline worlds/sc2/PoolFilter.py:136]: These checks are no longer necessary if the slot_data was updated earlier.

File: worlds/sc2/Rules.py [hunk 6]
```diff
@@ -925,7 +925,8 @@ def __init__(self, multiworld: MultiWorld, player: int):
         self.logic_level = get_option_value(multiworld, self.player, 'required_tactics')
         self.advanced_tactics = self.logic_level != RequiredTactics.option_standard
         self.take_over_ai_allies = get_option_value(multiworld, self.player, "take_over_ai_allies") == TakeOverAIAllies.option_true
-        self.kerrigan_unit_available = get_option_value(multiworld, self.player, 'kerrigan_presence') in kerrigan_unit_available
+        self.kerrigan_unit_available = get_option_value(multiworld, self.player, 'kerrigan_presence') in kerrigan_unit_available \
+            and SC2Campaign.HOTS in get_enabled_campaigns(multiworld, self.player)
```
Comment (MatthewMarinets) [inline worlds/sc2/Rules.py:929]: These checks are no longer necessary if the slot_data was updated earlier.

File: worlds/sc2/__init__.py [hunk 7]
```diff
@@ -432,7 +436,8 @@ def place_exclusion_item(item_name, location, locked_locations, player):
 def fill_pool_with_kerrigan_levels(multiworld: MultiWorld, player: int, item_pool: List[Item]):
     total_levels = get_option_value(multiworld, player, "kerrigan_level_item_sum")
     if get_option_value(multiworld, player, "kerrigan_presence") not in kerrigan_unit_available \
-            or total_levels == 0:
+            or total_levels == 0 \
+            or SC2Campaign.HOTS not in get_enabled_campaigns(multiworld, player):
```
Comment (MatthewMarinets) [inline worlds/sc2/__init__.py:440]: These checks are no longer necessary if the slot_data was updated earlier.

File: worlds/sc2/Locations.py [hunk 8]
```diff
@@ -78,7 +78,8 @@ def get_locations(multiworld: Optional[MultiWorld], player: Optional[int]) -> Tu
     # Note: rules which are ended with or True are rules identified as needed later when restricted units is an option
     logic_level = get_option_value(multiworld, player, 'required_tactics')
     adv_tactics = logic_level != RequiredTactics.option_standard
-    kerriganless = get_option_value(multiworld, player, 'kerrigan_presence') not in kerrigan_unit_available
+    kerriganless = get_option_value(multiworld, player, 'kerrigan_presence') not in kerrigan_unit_available \
+        or SC2Campaign.HOTS not in get_enabled_campaigns(multiworld, player)
```
Comment (MatthewMarinets) [inline worlds/sc2/Locations.py:82]: These checks are no longer necessary if the slot_data was updated earlier.

Comment (Magnemania) [reply worlds/sc2/Locations.py:82]: I believe slot data occurs after generation?  So it wouldn't be available at that point.

**Pull request merged**